test/test_util/fs_util.cpp

```diff
@@ -0,0 +1,42 @@
+#include "test_util/fs_util.h"
+
+#include <filesystem>
+#include <stdexcept>
+
+// Macros are not fun and should be avoided where possible, but
+// this check ensures that we fail at compile time in the event
+// that this definition is not present, rather than discovering
+// this at runtime and throwing an exception at test initialization.
+#ifndef NOISEPAGE_PROJECT_ROOT
+#error "NOISEPAGE_PROJECT_ROOT is not defined!"
+#endif
+
+// Necessary grossness for stringizing macro expansion.
+#define STR(x) #x
+#define XSTR(x) STR(x)
```

Comment:
Pick slightly longer names and undef at end of file?

Just in case it gets globbed into some other files with unity builds.

Reply:
Agreed. Updated.